src/components/feedback/Dialog.tsx

```diff
@@ -109,7 +118,41 @@ const DialogNext = function Dialog({
     } else {
       // Fall back to focusing the modal itself
       modalRef.current?.focus();
-      return;
+    }
+  }, [initialFocus, modalRef]);
+
+  const onTransitionEnd = (direction: 'in' | 'out') => {
+    if (direction === 'in') {
+      // We can't check the initial focus until transition "in" has finished
```

Comment:
See my above comment about perhaps renaming `setInitialFocus` to `initializeDialog` or similar. That would obviate the need for this comment.